src/JavascriptVM/vm.ts

```diff
@@ -0,0 +1,23 @@
+import Interpreter, { GlobalObject } from "js-interpreter";
+import { AppDispatch } from "../store";
+import { blocklySlice } from "../BlocklyInterface/blocklySlice";
+
+export const getIntepreter = (code: string, dispatch: AppDispatch) => {
+  function initInterpreter(
+    intepreter: Interpreter,
+    globalObject: GlobalObject
+  ) {
+    const highlightBlock = intepreter.createNativeFunction((id: string) => {
+      dispatch(blocklySlice.actions.highlightBlock({ blockId: id }));
```

Comment:
in https://github.com/FRUK-Simulator/blockly-intepretation-example/blob/master/src/interpreterwrapper.ts#L12 , there's an extra bit of state to track when the previous highlightBlock call was fired. This is so the [step button](https://github.com/FRUK-Simulator/blockly-intepretation-example/blob/master/src/interpreterwrapper.ts#L29) will step to the next blockly step, as opposed to the next intepreter step, meaning that hitting step will highlight the next block. It'd be nice to have that here as well

Reply:
Great feedback! I was trying to figure out how to correspond the interpreter code with the blockly code! Didn't dawn on me to just wait until the next block was highlighted!